includes/formFactory/ManageWikiFormFactoryBuilder.php

```diff
@@ -244,8 +248,11 @@ private static function buildDescriptorSettings(
 		$formDescriptor = [];
 
 		foreach ( $config->get( 'ManageWikiSettings' ) as $name => $set ) {
-			$add = ( $set['from'] == 'mediawiki' ) ||  in_array( $set['from'], $extList );
-			$disabled = ( $ceMW ) ? !( !$set['restricted'] || ( $set['restricted'] && $permissionManager->userHasRight( $context->getUser(), 'managewiki-restricted' ) ) ) : true;
+			$mwRequirements = isset( $set['requires'] ) ? ManageWikiRequirements::process( $set['requires'], $extList, false, $wiki ) : true;
+			$visible = isset( $set['requires']['visibility'] ) || isset( $set['requires']['extensions'] ) ? $mwRequirements : true;
```

Comment:
Why not just check requires? Or are there other parts of the array which should always return true even if set.

Also if an extension is not enabled, I feel like this should be disabled and it's default should match the extension's default - otherwise we'd have a setting in MWS which would be enabled, but it's enabled value would not be reflected in ManageWiki but hidden inside the extension itself.

Reply:
I think that should be done now